src/components/browse/browse.tsx

```diff
@@ -9,22 +9,20 @@ import { useSearchParams } from "react-router-dom";
 
 const Browse = () => {
   let [searchParams, setSearchParams] = useSearchParams();
-  let page = 1
-  let skip = 0;
-  const [filterDict, setFilterDict] = React.useState<facetFilterModel[]>([]);
+  const [page, setPage] = React.useState(parseInt(searchParams.get("p") || "0"))
   const [limit, setLimit] = React.useState(10);
-  const [searchKeyword, setSearchKeyword] = React.useState('');
+  let skip = page === 0 ? page * limit : (page - 1) * limit;
```

Comment:
Not sure why you do page * limit when page is 0, instead of just 0. Also, the second half should be page*limit for the skip, since pages are now starting from 0, otherwise page 1 and 2 (0 and 1 in code) would have the same skip (0 * limit, [1-1]*limit = 0*limit).

Reply:
The reason we have to calculate skip is because when the url contains a value for p (for eg browse/?p=8), we have to send the call with the appropriate skip value and not hardcoded 0.
The Browse page is not only the default page, it is also the redirected page for following routes
/browse/?p=n
/browse/?q=search&p=n
/browse/?f=filter&p=n
/browse/?q=search&f=filter&p=n

Reply:
That was not my concern. What I said is about the values. if page === 0 then page * limit is always 0, so why not just write 0? If page ===1, then (page -1) * limit will be 0 as well.

Reply:
done